Metalama.Framework.Engine/CodeModel/Builders/NamedTypeBuilder.cs

```diff
@@ -105,9 +106,13 @@ public bool IsPartial
     INamespace INamedType.ContainingNamespace => this.ContainingNamespace;
 
     public string FullName
-        => this.DeclaringType != null
-            ? $"{this.DeclaringType.FullName}.{this.Name}"
-            : $"{this.ContainingNamespace.FullName}.{this.Name}";
+        => this switch
+        {
+            { DeclaringType: not null } => $"{this.DeclaringType.FullName}.{this.Name}",
+            { ContainingNamespace.IsGlobalNamespace: true } => this.Name,
+            { ContainingNamespace.IsGlobalNamespace: false } => $"{this.ContainingNamespace.FullName}.{this.Name}",
+            _ => throw new AssertionFailedException( $"Unsupported: {this}" ),
```

Comment:
Nit: I don't think the default case is necessary here.

Reply:
Leaving it only for exception context in case ContainingNamespace is for some reason null.